trs_filer/ga4gh/trs/server.py

```diff
@@ -73,13 +73,27 @@ def toolsIdVersionsVersionIdGet(
         version with given id not found.
     """
 
-    obj = toolsIdGet.__wrapped__(id)
+    db_collection = (
+        current_app.config['FOCA'].db.dbs['trsStore']
+        .collections['objects'].client
+    )
 
-    for version in obj["versions"]:
-        if version['id'] == version_id:
-            return version
+    filt = {}
+    filt['id'] = id
+    filt['versions'] = {
+        '$elemMatch': {
+            'id': version_id
+        },
+    }
+    req_version = db_collection.find_one(
+        filter=filt,
+        projection={"_id": False},
+    )
+
+    if not req_version:
+        raise NotFound
 
-    raise NotFound
+    return req_version['versions'][0]
```

Comment:
@kushagra189  This will always return the first version in the array, *not* the version with the specified ID. The `$elemMatch` has to be specified in the projection instead of the filter. I have fixed this now, but the solution is still not very elegant. I think in principle the projection would allow you to return the exact document as we want it, without the need of subsetting `['versions'][0]` afterwards. You can open an issue for that, but set the priority to low, as it's really only to make the solution more elegant. As it is, it should work just fine.